### PR TITLE
Bluetooth: Mesh: Adds three Config Client API 

### PR DESCRIPTION
**Pull request by haoyanbiao**
for deleting appkey, unbinding an application from SIG model, unbinding an application from vendor model on the target node, withmatching shell command.
Signed-off-by: haoyanbiao <haoyanbiao@xiaomi.com>

### CODE REVIEW

File: include/bluetooth/mesh/cfg_cli.h [hunk 1]
```diff
@@ -273,6 +273,20 @@ int bt_mesh_cfg_app_key_add(uint16_t net_idx, uint16_t addr, uint16_t key_net_id
...
+ *  @param net_idx     Network index to encrypt with.
+ *  @param addr        Target node address.
+ *  @param key_net_idx Network key index the application key belongs to.
+ *  @param key_app_idx Application key index.
+ *  @param status      Status response parameter.
+ *
+ *  @return 0 on success, or (negative) error code on failure.
+ */
+int bt_mesh_cfg_app_key_del(u16_t net_idx, u16_t addr, u16_t key_net_idx,
+			    u16_t key_app_idx, u8_t *status);
```
Comment (trond-snekvik) [inline include/bluetooth/mesh/cfg_cli.h:288]: The Zephyr integer types have been deprecated, could you please convert these to stdint's `uint16_t` and `uint8_t`? Same throughout

File: subsys/bluetooth/mesh/cfg_cli.c [hunk 2]
```diff
@@ -1161,6 +1161,47 @@ int bt_mesh_cfg_app_key_get(uint16_t net_idx, uint16_t addr, uint16_t key_net_id
...
+	int err;
+
+	err = cli_prepare(&param, OP_APP_KEY_STATUS);
+	if (err) {
+		return err;
+	}
+
+	bt_mesh_model_msg_init(&msg, OP_APP_KEY_DEL);
+	key_idx_pack(&msg, key_net_idx, key_app_idx);
+	net_buf_simple_add_mem(&msg, app_key, 16);
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/cfg_cli.c:1188]: The key value field shouldn't be present in app_key_del.

File: subsys/bluetooth/mesh/shell.c [hunk 3]
```diff
@@ -1206,6 +1237,48 @@ static int cmd_mod_app_bind(const struct shell *shell, size_t argc,
...
+	if (status) {
+		shell_print(shell, "Model App Bind failed with status 0x%02x",
+			    status);
+	} else {
+		shell_print(shell, "AppKey successfully bound");
+	}
+
+	return 0;
+}
+
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/shell.c:1281]: Please add these two to the command list at the bottom, and add corresponding documentation entries in doc/reference/bluetooth/mesh/shell.rst :)

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
Thanks for contributing!

Looks good overall, but I have a couple comments that have to be resolved.

**Comment by trond-snekvik** [comment]
The Compliance check also has some issues with the format of your commit message.

The title needs to follow the <module>: <subject> format, ie "Bluetooth: Mesh: Add three Config Client APIs"

Additionally, the email in the signoff-line needs to match the email you committed with. Git can insert this automatically for you if you commit with the `-s` flag.

File: subsys/bluetooth/mesh/shell.c [hunk 4]
```diff
@@ -2603,6 +2676,8 @@ SHELL_STATIC_SUBCMD_SET_CREATE(mesh_cmds,
 		      0),
 	SHELL_CMD_ARG(app-key-add, NULL, "<NetKeyIndex> <AppKeyIndex> [val]",
 		      cmd_app_key_add, 3, 1),
+	SHELL_CMD_ARG(app-key-del, NULL, "<NetKeyIndex> <AppKeyIndex>",
+		      cmd_app_key_del, 2, 0),
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/shell.c:2680]: Have to include the `mesh` parameter in the count here, so the argument count parameter should be `3` here. The `mod-app-unbind` entry is correct.

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
Thanks. I just have one additional issue, and the compliance check has a couple of minor warnings that need to be resolved.

File: doc/reference/bluetooth/mesh/shell.rst [hunk 5]
```diff
@@ -413,6 +413,15 @@ The Configuration Client uses the general messages parameters set by ``mesh dst`
 	* ``NetKeyIndex``: Network key indexes to get a list of application key indexes from.
 
 
+``mesh app-key-del <NetKeyIndex> <AppKeyIndex>``
+------------------------------------------------------
+
+	Delete an application key from the target node.
+
+	* ``NetKeyIndex``: The network key index the application key is bound to.
+	* ``AppKeyIndex``: The application key index to add.
```
Comment (LingaoM) [inline doc/reference/bluetooth/mesh/shell.rst:422]: The application key index to delete.

**Review by LingaoM — CHANGES_REQUESTED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Review by LingaoM — APPROVED** [review]

File: include/bluetooth/mesh/cfg_cli.h [hunk 6]
```diff
@@ -287,6 +301,21 @@ int bt_mesh_cfg_app_key_get(uint16_t net_idx, uint16_t addr, uint16_t key_net_id
 int bt_mesh_cfg_mod_app_bind(uint16_t net_idx, uint16_t addr, uint16_t elem_addr,
 			     uint16_t mod_app_idx, uint16_t mod_id, uint8_t *status);
 
+/** @brief Unbind an application from a SIG model on the target node.
+ *
+ *  @param net_idx     Network index to encrypt with.
+ *  @param addr        Target node address.
+ *  @param elem_addr   Element address the model is in.
+ *  @param mod_app_idx Application index to bind.
```
Comment (LingaoM) [inline include/bluetooth/mesh/cfg_cli.h:309]: Application key index to unbind.

File: include/bluetooth/mesh/cfg_cli.h [hunk 7]
```diff
@@ -303,6 +332,22 @@ int bt_mesh_cfg_mod_app_bind_vnd(uint16_t net_idx, uint16_t addr, uint16_t elem_
 				 uint16_t mod_app_idx, uint16_t mod_id, uint16_t cid,
 				 uint8_t *status);
 
+/** @brief Unbind an application from a vendor model on the target node.
+ *
+ *  @param net_idx     Network index to encrypt with.
+ *  @param addr        Target node address.
+ *  @param elem_addr   Element address the model is in.
+ *  @param mod_app_idx Application index to bind.
```
Comment (LingaoM) [inline include/bluetooth/mesh/cfg_cli.h:340]: Same too.

File: include/bluetooth/mesh/cfg_cli.h (hunk 6 above)
Comment (LingaoM) [inline include/bluetooth/mesh/cfg_cli.h:309]: Application key index to unbind.

File: include/bluetooth/mesh/cfg_cli.h (hunk 7 above)
Comment (LingaoM) [inline include/bluetooth/mesh/cfg_cli.h:340]: Same too.

**Review by LingaoM — CHANGES_REQUESTED** [review]

**Review by LingaoM — APPROVED** [review]

**Pull request merged**